src/cursor.js

```diff
@@ -0,0 +1,753 @@
+    const bufferPosition = this.getBufferPosition()
+    const line = this.editor.lineTextForBufferRow(bufferPosition.row)
+    const firstCharacterColumn = line.search(/\S/)
+
+    if (firstCharacterColumn === -1) {
+      return false
+    } else {
+      return bufferPosition.column > firstCharacterColumn
+    }
+  }
+
+  // Public: Identifies if this cursor is the last in the {TextEditor}.
+  //
+  // "Last" is defined as the most recently added cursor.
+  //
+  // Returns a {Boolean}.
+  isLastCursor () {
+    return this === this.editor.getLastCursor()
+  }
+
+  /*
+  Section: Moving the Cursor
+  */
+
+  // Public: Moves the cursor up one screen row.
+  //
+  // * `rowCount` (optional) {Number} number of rows to move (default: 1)
+  // * `options` (optional) {Object} with the following keys:
+  //   * `moveToEndOfSelection` if true, move to the left of the selection if a
+  //     selection exists.
+  moveUp (rowCount = 1, {moveToEndOfSelection} = {}) {
+    let row, column
+    const range = this.marker.getScreenRange()
+    if (moveToEndOfSelection && !range.isEmpty()) {
+      ({row, column} = range.start)
+    } else {
+      ({row, column} = this.getScreenPosition())
+    }
+
+    if (this.goalColumn != null) column = this.goalColumn
+    this.setScreenPosition({row: row - rowCount, column}, {skipSoftWrapIndentation: true})
+    this.goalColumn = column
+  }
+
+  // Public: Moves the cursor down one screen row.
+  //
+  // * `rowCount` (optional) {Number} number of rows to move (default: 1)
+  // * `options` (optional) {Object} with the following keys:
+  //   * `moveToEndOfSelection` if true, move to the left of the selection if a
+  //     selection exists.
+  moveDown (rowCount = 1, {moveToEndOfSelection} = {}) {
+    let row, column
+    const range = this.marker.getScreenRange()
+    if (moveToEndOfSelection && !range.isEmpty()) {
+      ({row, column} = range.end)
+    } else {
+      ({row, column} = this.getScreenPosition())
+    }
+
+    if (this.goalColumn != null) column = this.goalColumn
+    this.setScreenPosition({row: row + rowCount, column}, {skipSoftWrapIndentation: true})
+    this.goalColumn = column
+  }
+
+  // Public: Moves the cursor left one screen column.
+  //
+  // * `columnCount` (optional) {Number} number of columns to move (default: 1)
+  // * `options` (optional) {Object} with the following keys:
+  //   * `moveToEndOfSelection` if true, move to the left of the selection if a
+  //     selection exists.
+  moveLeft (columnCount = 1, {moveToEndOfSelection} = {}) {
+    const range = this.marker.getScreenRange()
+    if (moveToEndOfSelection && !range.isEmpty()) {
+      this.setScreenPosition(range.start)
+    } else {
+      let {row, column} = this.getScreenPosition()
+
+      while (columnCount > column && row > 0) {
+        columnCount -= column
+        column = this.editor.lineLengthForScreenRow(--row)
+        columnCount--
+      } // subtract 1 for the row move
```

Comment:
It looks like `decaffeinate` put this comment on the wrong line. I think it should be on the previous line:

```javascript
  columnCount-- // subtract 1 for the row move
}
```

Reply:
Thanks!